### DISCUSSION TITLE
Adding Numeric Track number to File name

### DISCUSSION THREAD

**Discussion by DavidGreen63** [audiamus/AaxAudioConverter#88] [Ideas]
Hi Audiamus,
Although this fantastic application has a variety of file naming options, one that may prove useful is the option of adding the numeric track number to the file name. 
I know it is embedded in the meta data of the file, but its presence in the physical file name would benefit sorting in places where the meta data sort cannot be employed. And Audible just loved using track numbering in words, which works well as randomising tracks under an alphabetic sort.

**Comment by audiamus** [comment]
The option already exists, I believe. It's a combination of "file naming pattern" and "track numbering pattern". If you enabled named chapters, you can also set "always precede with number" in the Chapters tab in the Settings.  The outcome depends somewhat on conversion mode: single file, chapters, split chapters, etc., but chronological sorting by alpha-numeric file name should always be possible, I would think.
(Normally, sorting would be handled by the playlist, but I know there are audiobook players around that don't understand the playlist concept.)

**Reply by DavidGreen63** [reply]
I have only ever used Mode = Multiple files, by chapter; splitting up the AAX file into 1 file per chapter seemed logical, but Track numbering pattern is always greyed out (disabled) in that mode. 

I had never considered using the next Mode, ~, split into shorter tracks, as that would have split every chapter into tiny pieces. I suppose that doesn't make much difference so long as the split process doesn't disturb the content playback.

But I thinking of a File naming pattern of <#> - <track> - <book> - <author>. That is what I have now using a renaming utility to achieve, but the process is okay for a small list of files, painful for a typical title (35 chapters), but I know I'm going to hit titles with over 50 chapters in due course, and the renaming process is going to be daunting :)

But, you are right about the playlist, which would solve the problem; … but the application I use does not support playlists.

**Reply by audiamus** [reply]
>I suppose that doesn't make much difference so long as the split process doesn't disturb the content playback.

It definitely does not affect playback. Tracks will never be cut mid-word. It's meant to work like a music album, with similar short tracks, ideal for many car infotainment systems where you can skip back to the beginning of a track with the steering wheel controls. Very handy when distracted by traffic.

_To have a split chapter mode based on silence detection was one of the reasons why I created AAX Audio Converter in the first place._

>  That is what I have now using a renaming utility to achieve

As I said, have a look at the Basic Settings, page Chapters, and change "Use named chapters" from "Yes" to "Yes, always precede with number". That should do the trick for chapter mode.

**Reply by DavidGreen63** [reply]
Thanks for the tip of "always precede with number", that has eliminated the pre-playback preparation that I needed to do.
I've used the option for the last 2 audiobooks I have acquired, and has made the process so much faster.
The Tip, like the Application, is very much appreciated.
It is only because of this application that I even have an Audible account.

**Reply by nfriedly** [reply]
FYI: to get to settings you have to click the icon at the top left of the app. That opens a menu with a Basic Settings... option.
